packages/modules/product/src/repositories/product.ts

```diff
@@ -57,10 +58,18 @@ export class ProductRepository extends DALUtils.mikroOrmBaseRepositoryFactory(
     height?: string | number
     width?: string | number
   }) {
-    productToUpdate.weight = productToUpdate.weight?.toString()
-    productToUpdate.length = productToUpdate.length?.toString()
-    productToUpdate.height = productToUpdate.height?.toString()
-    productToUpdate.width = productToUpdate.width?.toString()
+    if (isDefined(productToUpdate.weight)) {
+      productToUpdate.weight = productToUpdate.weight?.toString()
+    }
+    if (isDefined(productToUpdate.length)) {
+      productToUpdate.length = productToUpdate.length?.toString()
+    }
+    if (isDefined(productToUpdate.height)) {
+      productToUpdate.height = productToUpdate.height?.toString()
+    }
+    if (isDefined(productToUpdate.width)) {
+      productToUpdate.width = productToUpdate.width?.toString()
+    }
```

Comment:
thought: we should probably find a better way to do this at some point 😄

can you create a ticket for it?